src/shared/components/common/button/RadioButton.tsx

```diff
@@ -0,0 +1,132 @@
+ * import { useState } from 'react';
+ * import RadioButton from '@/shared/components/common/button/RadioButton';
+ *
+ * const MyComponent = () => {
+ *   const [selected, setSelected] = useState('');
+ *   const options = [
+ *     { value: 'REJECTED', label: '거절' },
+ *     { value: 'INTERVIEW_PENDING', label: '면접대기' },
+ *     { value: 'INTERVIEW_COMPLETED', label: '면접 완료' },
+ *     { value: 'HIRED', label: '채용 완료', disabled: true },
+ *   ];
+ *
+ *   return (
+ *     <RadioButton
+ *       legend="현재 진행상태를 알려주세요."
+ *       name="applicationStatus"
+ *       options={options}
+ *       size="sm"
+ *       value={selected}
+ *       onChange={setSelected}
+ *     />
+ *   );
+ * };
+ * ```
+ *
+ * @example
+ * // 비활성화된 전체 그룹
+ * ```tsx
+ * <RadioButton
+ *   name="status"
+ *   options={options}
+ *   value={selected}
+ *   onChange={setSelected}
+ *   disabled={true}
+ * />
+ * ```
+ */
+const RadioButton = ({
+  name,
+  options,
+  value,
+  onChange,
+  disabled = false,
+  legend,
+  size = 'sm',
+}: RadioButtonProps): JSX.Element => {
+  const sizeStyle = {
+    sm: {
+      button: 'p-14 text-sm rounded-lg w-[327px]',
+      text: 'text-sm',
+    },
+    md: {
+      button: 'px-24 py-17 text-lg rounded-lg w-[360px]',
+      text: 'text-base',
+    },
+  };
+
+  const currentSize = sizeStyle[size];
+
+  return (
+    <fieldset>
+      {legend && <legend className='sr-only'>{legend}</legend>}
+      <div className='flex flex-col gap-8'>
+        {options.map((option) => {
+          const isDisabled = disabled || option.disabled;
+          const isChecked = value === option.value;
+          return (
+            <label
+              key={option.value}
+              className={` ${currentSize.button} flex cursor-pointer items-center justify-between border-2 font-medium transition-all duration-200 ${
+                isChecked ? 'border-mint-300 bg-gray-50' : 'border-line-100 hover:border-mint-300 bg-white'
+              } ${isDisabled ? 'cursor-not-allowed opacity-50' : ''} `}
+            >
+              <span
+                className={`${currentSize.text} ${isDisabled ? 'text-gray-400' : 'text-gray-600'} flex-1 select-none`}
+              >
+                {option.label}
+              </span>
+              <div className='relative inline-block'>
+                <input
+                  checked={isChecked}
+                  className='w-22 h-22 checked:border-mint-300 block cursor-pointer appearance-none rounded-full border border-gray-200 bg-white transition-all duration-200 checked:bg-white disabled:cursor-not-allowed disabled:opacity-50'
+                  disabled={isDisabled}
+                  name={name}
+                  type='radio'
+                  value={option.value}
+                  onChange={(e) => {
+                    onChange(e.target.value);
+                  }}
+                />
+                {isChecked && <div className='bg-mint-300 absolute left-6 top-6 h-10 w-10 rounded-full' />}
```

Comment:
```suggestion
                {isChecked && <div className='bg-mint-300 absolute left-6 top-6 size-10 rounded-full' />}
```
width와 height가 같다면 size로 작성해도 좋을 거 같습니다!